src/Controller/ControllerInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace RebelCode\EddBookings\RestApi\Controller;
+
+use RebelCode\EddBookings\RestApi\Resource\ResourceInterface;
+
+/**
+ * An API controller - something that can work with API resources.
+ *
+ * @since [*next-version*]
+ */
+interface ControllerInterface
+{
+    /**
+     * Retrieves resources.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $params The params.
```

Comment:
Must this be a list, a container, or both?

Reply:
Any kind of container.

I am aware that some classes/methods depend only on array and use array syntax. I've been meaning to go through and amend everything as necessary such that all parameters document more flexible types and that the classes use the container helper traits.